rs/cross-chain/proposal-cli/src/dashboard/mod.rs

```diff
@@ -0,0 +1,35 @@
+//! Retrieve data from the public ICP dashboard REST API
+//! https://ic-api.internetcomputer.org/api/v3/swagger
+mod responses;
+#[cfg(test)]
+mod tests;
+
+use crate::dashboard::responses::CanisterInfo;
+use candid::Principal;
+use std::collections::BTreeSet;
+
+pub struct DashboardClient {
+    client: reqwest::Client,
+    base_url: String,
+}
+
+impl DashboardClient {
+    pub fn new() -> Self {
+        Self {
+            client: reqwest::Client::new(),
+            base_url: "https://ic-api.internetcomputer.org/api/v3".to_string(),
```

Comment:
Just wonder if the canister_info endpoint here is any different than the canister_info exposed from IC management canister.. and curious how it is implemented because the latter can only be called from a canister.

Reply:
The dashboard [API](https://ic-api.internetcomputer.org/api/v3/swagger) at `api/v3/canisters/{canister_id}` is I would say quite different from `canister_info` from the [management canister](https://internetcomputer.org/docs/current/references/ic-interface-spec#ic-canister-info):

1. the public dashboard maintains its own centralized database, recording for example all proposals.
2. This allows to offer via its public API much more information (albeit in a centralized manner) than what `canister_info` gives back, e.g. IDs of upgrade proposals for a given canister ID.

Reply:
Fee free to ping me if there is anything left opened. Marking as resolved to merge.